unfolding/1D/Bayes/runCorrectionChain1DBayes_SpectrumTaskSimplePoor_8TeV.cpp

```diff
@@ -0,0 +1,446 @@
+    auto hspecandclass = static_cast<TH2 *>(histos->FindObject("hJetSpectrumAbs"));
+    auto rawspectrum = hspecandclass->ProjectionY(histnamebuilder.str().data(), clusterbin, clusterbin);
+    rawspectrum->SetDirectory(nullptr);
+
+    // calculate norm
+    auto hnorm = static_cast<TH1 *>(histos->FindObject("hClusterCounterAbs"));
+    auto norm = hnorm->GetBinContent(clusterbin);
+    // calculate bin0 correction
+    auto evhist = static_cast<TH1 *>(histos->FindObject("fNormalisationHist")); // take the one from the AliAnalysisTaskEmcal directly
+    auto bin0correction = evhist->GetBinContent(evhist->GetXaxis()->FindBin("Event selection")) / evhist->GetBinContent(evhist->GetXaxis()->FindBin("Vertex reconstruction and quality"));
+    norm *= bin0correction;
+
+    return {norm, rawspectrum};
+}
+
+TH2 *getResponseMatrix(TFile &reader, double R, const std::string_view sysvar, int closurestatus = 0) {
+    std::string closuretag;
+    switch(closurestatus) {
+    case 0: break;
+    case 1: closuretag = "Closure"; break;
+    case 2: closuretag = "NoClosure"; break;
+    };
+    std::string responsematrixbase = "hJetResponseFine";
+    std::stringstream responsematrixname;
+    responsematrixname << responsematrixbase;
+    if(closuretag.length()) responsematrixname <<  closuretag;
+    std::stringstream dirnamebuilder;
+    dirnamebuilder << "EnergyScaleResults_FullJet_R" << std::setw(2) << std::setfill('0') << int(R*10.) << "_INT7";
+    if(sysvar.length()) dirnamebuilder << "_" << sysvar;
+    reader.cd(dirnamebuilder.str().data());
+    auto histlist = static_cast<TKey *>(gDirectory->GetListOfKeys()->At(0))->ReadObject<TList>();
+    auto rawresponse = static_cast<TH2 *>(histlist->FindObject(responsematrixname.str().data()));
+    rawresponse->SetDirectory(nullptr);
+    rawresponse->SetNameTitle(Form("Rawresponse_R%02d", int(R*10.)), Form("Raw response R=%.1f", R));
+    return rawresponse;
+}
+
+double getCENTNOTRDCorrection(TFile &reader, const std::string_view sysvar){
+    std::stringstream dirnamebuilder;
+    dirnamebuilder << "JetSpectrum_FullJets_R02_EJ1";
+    if(sysvar.length()) dirnamebuilder << "_" << sysvar;
+    reader.cd(dirnamebuilder.str().data());
+    auto histos = static_cast<TKey *>(gDirectory->GetListOfKeys()->At(0))->ReadObject<TList>();
+    auto hnorm = static_cast<TH1 *>(histos->FindObject("hClusterCounter"));
+    return hnorm->GetBinContent(3)/hnorm->GetBinContent(2);
+}
+
+std::pair<double, TH1 *> makeRejectionFactor(TH1 *histhigh, TH1 *histlow, double R, const std::string_view triggerlow, const std::string_view triggerhigh, const std::string_view sysvar, double triggerswap){
+    auto rfac = histcopy(histhigh);
+    rfac->SetNameTitle(Form("RejecionFactor_%s/%s_R%02d", triggerhigh.data(), triggerlow.data(), int(R * 10.)), Form("Rejection Factor for %s/%s for R=%.1f", triggerhigh.data(), triggerlow.data(), R));
+    rfac->SetDirectory(nullptr);
+    rfac->Divide(histhigh, histlow, 1., 1.);
```

Comment:
Usually the rejection factor is calculated via event-normalised spectra. However you need to make a local copy and not apply on the original, in order to not apply it twice.